packages/design-system/src/components/tree/tree-node.tsx

```diff
@@ -0,0 +1,306 @@
+      if (event.target === collapsibleContentRef.current) {
+        onExpandTransitionEnd?.();
+      }
+    },
+    [onExpandTransitionEnd]
+  );
+
+  // If ther's no animation, we need to manually trigger onExpandTransitionEnd
+  const prevIsExpanded = useRef(isExpanded);
+  useEffect(() => {
+    if (animate === false && isExpanded !== prevIsExpanded.current) {
+      onExpandTransitionEnd?.();
+    }
+    prevIsExpanded.current = isExpanded;
+  }, [animate, onExpandTransitionEnd, isExpanded]);
+
+  return (
+    <Collapsible.Root
+      open={isExpanded}
+      onOpenChange={(isOpen) => setIsExpanded?.(itemData, isOpen)}
+      data-drop-target-id={itemData.id}
+    >
+      <ItemWrapper
+        isSelected={isSelected}
+        parentIsSelected={parentIsSelected}
+        enableHoverState={forceHoverStateAtItem === undefined}
+        forceHoverState={forceHoverStateAtItem === itemData.id}
+      >
+        {/* We want the main ItemButton to take the entire space,
+         * and then position the collapsible trigger on top of it using absolute positionning.
+         * When user clicks anywhere on a tree item, they should either hit the main button or the trigger.
+         */}
+
+        <ItemButton
+          type="button"
+          data-drag-item-id={itemData.id}
+          data-item-button-id={itemData.id}
+          onFocus={makeSelected}
+        >
+          <NestingLines isSelected={isSelected} level={level} />
+          {isAlwaysExpanded === false && <TriggerPlaceholder />}
+          {renderItem({ data: itemData, isSelected })}
+        </ItemButton>
+
+        {shouldRenderExpandButton && (
+          <CollapsibleTrigger
+            style={{ left: (level - 1) * INDENT + ITEM_PADDING }}
+            // We don't want a separate focusable control inside a tree item.
+            // tabIndex makes it skipped over when tabbing.
+            // It still can be focused using mouse, but we handle this elsewhere.
+            tabIndex={-1}
+          >
+            {isExpanded ? <TriangleDownIcon /> : <TriangleRightIcon />}
+          </CollapsibleTrigger>
+        )}
+      </ItemWrapper>
+      {isExpandable && (
+        <CollapsibleContent
+          onAnimationEnd={handleAnimationEnd}
+          ref={collapsibleContentRef}
+        >
+          {
+            // CollapsibleContent doesn't render children when collapsed.
+            // No need to worry about optimizing this.
+            itemChildren.map((child) => (
+              <TreeNode
+                key={child.id}
+                itemData={child}
+                level={level + 1}
+                parentIsSelected={isSelected || parentIsSelected}
+                {...commonProps}
+              />
+            ))
+          }
+        </CollapsibleContent>
+      )}
+    </Collapsible.Root>
+  );
+};
+
+const Label = styled(__DEPRECATED__Text, {
+  whiteSpace: "nowrap",
+  overflow: "hidden",
+  textOverflow: "ellipsis",
+  lineHeight: 1.4,
```

Comment:
why do we need line height here? isn't it a single line text and should be 1? I assume you are using it for padding and for that we should use padding, line-height is for distance within multiline text

Reply:
hmm, it shouldn't, lets check

Reply:
Added a code comment